libs/execution_context/ebpf_maps.c

```diff
@@ -801,10 +801,9 @@ _get_object_from_array_map_entry(_Inout_ ebpf_core_map_t* map, _In_ const uint8_
         ebpf_object_type_t value_type =
             (map->ebpf_map_definition.type == BPF_MAP_TYPE_PROG_ARRAY) ? EBPF_OBJECT_PROGRAM : EBPF_OBJECT_MAP;
         if (id != 0) {
-
-            // Note that this call might fail and that's fine.  The id might be valid, but the object might have been
-            // since deleted.
-            (void)EBPF_OBJECT_REFERENCE_BY_ID(id, value_type, &object);
+            if (ebpf_object_pointer_by_id(id, value_type, &object) != EBPF_SUCCESS) {
+                object = NULL;
```

Comment:
Is it needed to set object to NULL in the failure case? (It looks like it would have remained NULL if ebpf_object_pointer_by_id returns error).